src/components/TalentPage/components/ListProofs/components/ProofBlock/components/Kudos.jsx

```diff
@@ -0,0 +1,63 @@
+import s from "./Kudos.module.scss";
+import { useState, useContext, useEffect } from "react";
+import { UserContext } from "../../../../../../../context/UserContext";
+import { TalentsService } from "../../../../../../../services/api-services";
+import { TalentsContext } from "../../../../../../../context/TalentsContext";
+
+export function Kudos({ id }) {
+    const [kudos, setKudos] = useState(0);
+    const [isLiked, setIsLiked] = useState(false);
+    const { token } = useContext(UserContext);
+    const { page, size } = useContext(TalentsContext);
+    const { talentsProofs } = useContext(UserContext);
+    const { user } = useContext(UserContext);
+
+    useEffect(() => {
+        TalentsService.getKudos(id, token)
+            .then((kudos) => {
+                setKudos(kudos.amount);
+                setIsLiked(kudos.has_kudos);
+            })
+            .catch((err) => console.log(err));
+    }, []);
+
+    function likeProof() {
+        if (token) {
+            if (!isLiked) {
+                TalentsService.putKudos(id, token)
+                    .then((response) => {
+                        setKudos(kudos + 1);
+
+                        setIsLiked(!isLiked);
```

Comment:
You should use previous state
```suggestion
                        setIsLiked((prev)=>!prev);
```